packages/amplify_datastore/android/src/test/kotlin/com/amazonaws/amplify/amplify_datastore/AmplifyModelSchemaTest.kt

```diff
@@ -0,0 +1,339 @@
+import org.mockito.Mockito.verify
+import org.mockito.invocation.InvocationOnMock
+import org.robolectric.RobolectricTestRunner
+
+// test that map received then transformed to flutter equivalent
+// generates appropriate native equivalent
+
+
+
+@RunWith(RobolectricTestRunner::class)
+class AmplifyModelSchemaTest {
+
+    @Test
+    fun test_BlogSchema() {
+
+        // Generate Flutter ModelSchema from map input
+        var inputMap : Map<String, Any> =
+                mapOf(
+                        "name" to "Blog",
+                        "pluralName" to "Blogs",
+                        "authRules" to emptyList<Any>(),
+                        "fields" to mapOf<String,Any>(
+                                "id" to mapOf(
+                                        "name" to "id",
+                                        "type" to
+                                                mapOf(
+                                                        "fieldType" to 0,
+                                                        "ofModelName" to ""),
+                                        "isRequired" to true,
+                                        "isArray" to false,
+                                        "association" to mapOf<String,Any>(),
+                                        "authRules" to emptyList<Any>()
+                                ),
+                                "name" to mapOf(
+                                        "name" to "name",
+                                        "type" to mapOf(
+                                                "fieldType" to 0,
+                                                "ofModelName" to ""),
+                                        "isRequired" to true,
+                                        "isArray" to false,
+                                        "association" to mapOf<String,Any>(),
+                                        "authRules" to emptyList<Any>()
+                                ),
+                                "posts" to mapOf(
+                                        "name" to "posts",
+                                        "type" to mapOf(
+                                                "fieldType" to 10,
+                                                "ofModelName" to "Post"),
+                                        "isRequired" to false,
+                                        "isArray" to true,
+                                        "association" to mapOf(
+                                                "associationType" to 0,
+                                                "targetName" to "",
+                                                "associatedName" to "blog",
+                                                "associatedType" to "Blog"
+                                        ),
+                                        "authRules" to emptyList<Any>()
+                                )
+                        )
+                )
```

Comment:
Using the same file as dart unit tests would probably require us to make a shared json file.  This would require investigating serializing from json to the same map format in Kotlin.  

I've looked online for some tools but don't know which one will actually work as expected and I'm hesitant to spend too much time on this point right now. 

@jamesonwilliams do you have any suggestions for easily loading a json file as a Kotlin map?

Reply:
Query tests use extensive use of reading from json files and converting to a kotlin map. Is it not sufficient?